DeepLink-org/InternLM
# llj dev

- feat: add unitest for model (#300)
- Merge main to develop (#309)
- docs(doc/code-docs): add figure for training docs (#307)
- Merge main to develop (#312)
- feat: more tgs (#310)
- feat: add optimizer_unitest (#303)
- feat(tools): support openai api (#313)
- fix(tools): fix yield bug in stream_chat (#315)
- test(tests/test_training): add training e2e tests for loss spike and loss accuracy (#304)
- feat(linear): optimize mlp by using jit  (#321)
- fix(.github/workflows/e2e_test.yaml): update ci runner name
- feat(mdoel): add DynamicNTKScalingRotaryEmbedding (#339)
- fix(model/embedding.py): ci lint check error (#345)
- Feat(PythonGC): Do garbage collection manually (#326)
- feat(.github/workflows): update ci e2e tests and add ci unit tests (#324)
- feat(doc/code-docs): add checkpoint save/load usage doc (#311)
- fix(storage): fix try_get_storage_backend (#359)
- fix(config): monitor config key error when args_check is False (#362)
- fix(monitor): add volc and aliyun jobid (#338)
- doc(monitor): add light monitoring doc (#352)
- feat(numa): bind numa if possible (#320)
- doc(usage): add dynamic ntk into doc (#367)
- Feat(norm)/support fused precision (#319)
- docs(doc/code-docs): update document image for InternLM parallel architecture (#373)
- feat(moe): add moe module (#182)
- docs(doc): add tf32 docs (#374)
- fix the moe loss as none for panel_metrics (#379)
- feat(moe): add  local data parallel support for experts (#376)
- Doc(config): add auto_resume annotation into example config (#380)
- fix(doc): fix huggingface url (#392)
- make seed in different tensor rank different (#405)
- fix(pipeline): fix bugs for pipeline when enable mixed precision (#382)
- feat(moe):support zero for expert local dp (#404)
- feat(train): add fsdp training option (#293)
- fix(model): fix errant inference_forward (#396)
- fix(optimizer/fsdp_optimizer.py): fsdp process empty params group (#408)
- feat(storage): support volc oss ckpt saving (#397)
- fix(init): allow resume_tb_folder is an empty string (#391)
- fix(moe): fix moe compatibility for fsdp and memory profiling (#417)
- compatible with old ckpt (#418)
- Test(pp): test pipeline parallel (#413)
- feat(monitor): send exception to light monitor (#420)
- fix(utils): disable bench_net in gputest.py (#421)
-  Doc(moe): add documentation for moe training (#411)
- fix: unitest (#424)
- fix(pipeline): fix interleave type assert and metrics error (#423)
- test(workflow): add unit test yaml (#427)
- feat(optimizer): add layer norm to tensorboard (#429)
- fix(*)/all-reduce for norm in sequence parallel (#443)
- fix broadcast synchronize() (#450)
- fix(optimizer):broadcast (#453)
- feat(optimizer): zero gradient count (#449)
- volc_path (#454)
- feat(storage): support ali oss ckpt saving (#439)
- fix(os): fix FileNotFoundError in storage_manager (#455)
-  fix(moe): remove norm&gate force sync (#448)
- change slurm partition (#464)
- send exception to light monitor only if the server is available (#465)
- refactor grad norm profiling (#466)
- fix(tools): fix streaming_chat and update docs (#467)
- fix(readme): fix model path in readme (#474)
- feat(tools): support dynamic ntk rope in transformers (#470)
- fix(data): broadcast list when walking folders (#475)
- feat(data): walk folder to get dataset_type_ids_map (#477)
- fix(tools): set bos, eos, pad in convert2hf to fix improper generation (#471)
- feat(doc): add dynamic ntk example (#480)
- feat(doc): add torch_dtype to examples in README (#479)
- Feat(QA): Check model weights for acc (#476)
- feat(tools): support origin internlm architecture in web_demo (#478)
- init light monitoring on all ranks (#462)
- rename vars (#468)
- fix(metric): use float32 to compute ppl (#481)
- feat(ckpt): save ckpt when reach total step count (#486)
- docs(code-docs): add 20b training demo (#488)
- fix dataset types when using random dataset (#489)
- fix(docs): fix 20B demo log (#490)
- fix(alert): send exception of all ranks (#491)
- fix(train): unify the exp paths (#492)
- test(workflow): add ci workflow for acc test (#485)
- feat(train): update get_train_data_loader to make logic clearer (#498)
- Feat(QA): Check init model weights (#502)
- unitest_only_forward (#484)
- feat(train): support_rampup_batch_size and fix bugs (#493)
- test(workflow): add model init test (#504)
- feat(seed): set global seed for every model initialization (#496)
- fix(timeout): larger timeout (#495)
- feat(doc): add GPU memory info for 7B & 20B models (#507)
- feat(model): add rope_base interface (#512)
- Feat(QA): Check loss when swapping micro_num and micro_bsz && Check grad norm (#510)
- name (#514)
- 1. fix(config): rampup_batch_size defalut value BC. (#515)
- test(workflow): add workflow for loss test and change trigger event (#513)
- fix(ci): fix test model ckpt ci test (#518)
- add unit test case (#524)
- feat(storage): use multipart upload when using oss (#520)
- fix test_checkpoint (#526)
- fix(model): add IS_SEQUENCE_PARALLEL check for norm module (#528)
- add output embedding tf32 option (#523)
- feat(grad_norm): vocab grad norm profiling (#519)
- fix the type_ids when micro_num=1 and use_flash_attn=False (#516)
- fix(storage): unify the name of ak & sk (#527)
- fix(test): fix type_ids unpack bug (#530)
- feat(model): support llama model with checkpoint loading (#532)
- fix(metric): add metric dtype control (#533)
- feat(ckpt): support auto resume in Volc and Ali (#529)
- fix the bug so that the sequence parallel norm is all-reduced when overlap is False (#534)
- fix(pp): fix no-packed dataset load micro batch error (#538)
- fix(model): change model_type `LLAMA` to `LLAMA2` (#539)
- fix(moe): fix moe zero mode bug (#548)
- feat: try supporting dipu in 7B
- feat: support 30
- fix: add a missing config
- 30B_llama2_nockpt config & remove nvcc compilation dependencies in third_party
